xbmc/video/dialogs/GUIDialogVideoVersion.h

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "MediaSource.h"
+#include "guilib/GUIDialog.h"
+#include "media/MediaType.h"
+#include "video/VideoDatabase.h"
+
+#include <memory>
+
+class CFileItem;
+class CFileItemList;
+class CVideoDatabase;
+
+class CGUIDialogVideoVersion : public CGUIDialog
+{
+public:
+  CGUIDialogVideoVersion(void);
+  ~CGUIDialogVideoVersion(void) override;
+  bool OnMessage(CGUIMessage& message) override;
+  void SetVideoItem(const CFileItem *item, bool playMode = false);
+  static int NewVideoVersion(std::string& strVideoVersion);
+  static bool SetVideoVersion(const CFileItem* item);
+  static bool ProcessVideoVersion(VideoDbContentType iType, int dbId);
+  static int SelectVideoVersion(const CFileItem* item);
+  static void ShowVideoVersion(const CFileItem* item, bool playMode =  false);
+  static std::string GenerateExtrasVideoVersion(const std::string& strExtrasRoot, const std::string& strExtrasPath);
+  static std::string GenerateExtrasVideoVersion(const std::string& strExtrasPath);
+
+protected:
+  void OnInitWindow() override;
+  void ClearVideoVersionList();
+  void SetDefaultVideoVersion();
+  void RemoveVideoVersion();
+  void RemoveVideoExtras();
+  void RemoveVideoVersionItemFromList(CFileItemList* list, CFileItem* item);
+  void AddVideoVersion(bool primary = true);
+  void ChooseArt();
+  void Play();
+
+  std::shared_ptr<CFileItem> m_videoItem;
+  bool m_playMode;
```

Comment:
Why are these methods and members protected, not private? Do you expect somebody deriving from this class?

Reply:
Indeed, PoLP - Principle of Least Privelege - always prefer private unless the design actively requires protected.